server/src/main/java/org/apache/uniffle/server/storage/LocalStorageManager.java

```diff
@@ -143,7 +143,7 @@ public Storage selectStorage(ShuffleDataFlushEvent event) {
         event.getStartPartition()));
     if (storage.containsWriteHandler(event.getAppId(), event.getShuffleId(), event.getStartPartition())
         && storage.isCorrupted()) {
-      throw new RuntimeException("storage " + storage.getBasePath() + " is corrupted");
```

Comment:
Why do we remove this exception?

Reply:
I don't know what is the purpose of this exception. And i think it should continue if one storage is corrupted.

Reply:
The removal of the Exception seems irreverent with the title, can you revert this change?
In another word, you should change it in another PR if it's really needed.

Reply:
`DiskErrorToleranceTest#diskErrorTest` will fail without this change. https://github.com/apache/incubator-uniffle/actions/runs/3328036207/jobs/5503556615

Reply:
OK, it makes sense.

Reply:
@jerqi will this change affect other code path? (seems fine in CI)

Reply:
I need to ensure the influence.

Reply:
This exception will help application fast fail. If a storage which ever was written with the application data, although we write new data to another storage, they will also fail. So we throw an exception here,.

Reply:
> This exception will help application fast fail. If a storage which ever was written with the application data, although we write new data to another storage, they will also fail. So we throw an exception here,.

Is it better to throw a different kind of Exception here such as `IllegalStateException` and catch it in the test (where the Exception is expected).

Reply:
i think we should consider whether it is reasonable first.

Reply:
> This exception will help application fast fail. If a storage which ever was written with the application data, although we write new data to another storage, they will also fail. So we throw an exception here,.

Actually application seems will not fast fail even if the exception be throw, it will be catch in `ShuffleFlushManager`. And if   we let it continue, we just lost the data in the corrupted disk, but in original logic we will lost all data of this replica. It will help after #276

Reply:
As I know, the data wont be lost when selecting a corrupted disk, just being preserved in the memory.

Reply:
> As I know, the data wont be lost when selecting a corrupted disk, just being preserved in the memory.

Is this more serious?

Reply:
That means this exception thrown is OK for the app.

Reply:
I'm afraid not. many data will be preserved in memory and not released.

Reply:
> `DiskErrorToleranceTest#diskErrorTest` will fail without this change. https://github.com/apache/incubator-uniffle/actions/runs/3328036207/jobs/5503556615

But how do I deal with this problem. I think this ut is reasonable. Merge another pr first?

Reply:
@jerqi What do you think? fix in another pr or in current pr? if fix in another pr, i will fix `DiskErrorToleranceTest#diskErrorTest` in current pr.

Reply:
How do we determine that disk is broken?
I think the disk can't be read or written. So I don't think this is a bug.
So we should let the application fail fast. But I already have multiple replicas, so it's unnecessary to fail fast here. 
It's ok for me after I think twice.